packages/cdktf/lib/terraform-local.ts

```diff
@@ -22,31 +22,31 @@ export class TerraformLocal
   }
 
   public get expression() {
-    return Token.asAny(ref(this.interpolation()));
+    return Token.asAny(this.interpolation());
   }
 
   public get asString(): string {
-    return Token.asString(ref(this.interpolation()));
+    return Token.asString(this.interpolation());
   }
 
   public get asNumber(): number {
-    return Token.asNumber(ref(this.interpolation()));
+    return Token.asNumber(this.interpolation());
   }
 
   public get asList(): string[] {
-    return Token.asList(ref(this.interpolation()));
+    return Token.asList(this.interpolation());
   }
 
   public get asBoolean(): IResolvable {
-    return ref(this.interpolation());
+    return this.interpolation();
   }
 
   private interpolation(): any {
-    return `local.${this.friendlyUniqueId}`;
+    return ref(`local.${this.friendlyUniqueId}`);
   }
 
   public get fqn() {
-    return `local.${this.friendlyUniqueId}`;
+    return Token.asString(`local.${this.friendlyUniqueId}`);
```

Comment:
I wonder if we should wrap this with ref, otherwise there could be an extra "${}" if this is used within a Terraform function

Reply:
I could see us wanting to have some more information when working on cross-stack references, but at this time there is no `$` present in the string, so I don't there will be any problems there.

Reply:
Oh yeah, I missed that, makes sense 👍